src/beautifiers/typescript-formatter.coffee

```diff
@@ -8,28 +9,53 @@ module.exports = class TypeScriptFormatter extends Beautifier
     TypeScript: true
   }
 
-  beautify: (text, language, options) ->
-    return new @Promise((resolve, reject) =>
+  beautify: (text, language, options, {filePath}) ->
+    try
+      base = require("typescript-formatter/lib/provider/base").default
+      tsconfigjson = require("typescript-formatter/lib/provider/tsconfigjson").default
+      { default: editorconfig, postProcess: editorconfigPostProcess } = require("typescript-formatter/lib/provider/editorconfig")
+      { default: tslintjson, postProcess, tslintPostProcess } = require("typescript-formatter/lib/provider/tslintjson")
 
-      try
-        format = require("typescript-formatter/lib/formatter").default
-        formatterUtils = require("typescript-formatter/lib/utils")
-        # @verbose('format', format, formatterUtils)
+      format = require("typescript-formatter/lib/formatter").default
+      formatterUtils = require("typescript-formatter/lib/utils")
+      # @verbose('format', format, formatterUtils)
 
-        opts = formatterUtils.createDefaultFormatCodeOptions()
+      optionModifiers = [base, tsconfigjson, editorconfig, tslintjson, pluginOptsModifier options]
+      postProcessors = [editorconfigPostProcess, tslintPostProcess, fileEndingPostProcess]
 
-        if options.indent_with_tabs
-          opts.ConvertTabsToSpaces = false
-        else
-          opts.TabSize = options.tab_width or options.indent_size
-          opts.IndentSize = options.indent_size
-          opts.IndentStyle = 'space'
+      formatSettings = formatterUtils.createDefaultFormatCodeOptions()
 
-        @verbose('typescript', text, opts)
-        result = format('', text, opts)
-        @verbose(result)
-        resolve result
-      catch e
-        return reject(e)
+      processFormatCodeOptions(optionModifiers, filePath, {}, formatSettings)
+        .then((formatSettings) =>
+          @verbose('typescript', text, formatSettings)
+          result = format('', text, formatSettings)
+          @verbose(result)
+          return result
+        )
+    catch e
+      return Promise.reject(e)
 
-    )
+processFormatCodeOptions = (optionModifiers, fileName, opts, formatSettings) ->
+  modifiers = optionModifiers.slice 0
+
+  next = (_formatSettings) ->
+    if modifiers.length is 0
+      return Promise.resolve(formatSettings)
+    modifier = modifiers.shift()
+    ret = modifier(fileName, opts, formatSettings)
+    return Promise.resolve(ret).then((formatSettings) -> next formatSettings)
+
+  next formatSettings
+
+fileEndingPostProcess = (fileName, formattedCode, opts, formatSettings) ->
+  Promise.resolve formattedCode.replace(/\r?\n/g, formatSettings.NewLineCharacter or '\n')
```

Comment:
I'm not sure where `NewLineCharacter` comes from, however you should use the Atom-Beautify configured value.

See:
- https://github.com/Glavin001/atom-beautify/blob/c881bdf656f6ab3daf5516209d3960f97ba1ab83/src/languages/javascript.coffee#L106-L110
- https://github.com/Glavin001/atom-beautify/blob/0f691d5116e21dac38fb75636197c24e4b173968/src/beautifiers/js-beautify.coffee#L30
- https://github.com/Glavin001/atom-beautify/pull/1469

Then be sure to generate the new documentation with `npm run docs`.